src/Listener/JsonApiListener.php

```diff
@@ -0,0 +1,264 @@
+<?php
+namespace Crud\Listener;
+
+use Cake\Core\Configure;
+use Cake\Event\Event;
+use Cake\Network\Exception\BadRequestException;
+use Cake\Utility\Inflector;
+use Crud\Error\Exception\CrudException;
+use Crud\Event\Subject;
+use Crud\Traits\QueryLogTrait;
+
+/**
+ * Extends Crud ApiListener to respond in JSON API format.
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ */
+class JsonApiListener extends ApiListener
+{
+    use QueryLogTrait;
+
+    /**
+     * Default configuration
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'detectors' => [
+            'jsonapi' => ['ext' => 'json', 'accepts' => 'application/vnd.api+json'],
+        ],
+        'exception' => [
+            'type' => 'default',
+            'class' => 'Cake\Network\Exception\BadRequestException',
+            'message' => 'Unknown error',
+            'code' => 0,
+        ],
+        'exceptionRenderer' => 'Crud\Error\JsonApiExceptionRenderer',
+        'setFlash' => false,
+        'urlPrefix' => null, // string holding URL to prefix links in jsonapi response with
+        'withJsonApiVersion' => false, // true or array/hash with additional meta information (will add top-level node `jsonapi` to the response)
+        'meta' => false, // array or hash with meta information (will add top-level node `meta` to the response)
+        'include' => [],
+        'fieldSets' => [], // hash to limit fields shown (applicable to both `data` and `included` nodes)
+    ];
+
+    /**
+     * Returns a list of all events that will fire in the controller during its lifecycle.
+     * You can override this function to add you own listener callbacks
+     *
+     * We attach at priority 10 so normal bound events can run before us
+     *
+     * @return array
+     */
+    public function implementedEvents()
+    {
+        $this->setupDetectors();
+
+        // Accept body data posted with Content-Type `application/vnd.api+json`
+        $this->_controller()->RequestHandler->config([
+            'inputTypeMap' => [
+                'jsonapi' => ['json_decode', true]
+            ]
+        ]);
+
+        return [
+            'Crud.beforeFilter' => ['callable' => [$this, 'setupLogging'], 'priority' => 1],
+            'Crud.beforeHandle' => ['callable' => [$this, 'beforeHandle'], 'priority' => 10],
+            'Crud.setFlash' => ['callable' => [$this, 'setFlash'], 'priority' => 5],
+            'Crud.beforeRender' => ['callable' => [$this, 'respond'], 'priority' => 100],
+            'Crud.beforeRedirect' => ['callable' => [$this, 'respond'], 'priority' => 100]
+        ];
+    }
+
+    /**
+     * beforeHandle
+     *
+     * Called before the crud action is executed.
+     *
+     * @param \Cake\Event\Event $event Event
+     * @return void
+     */
+    public function beforeHandle(Event $event)
+    {
```

Comment:
for usability, it would be good service to check here if `neomerx/json-api` is installed or not, and throw an exception in case it isn't ?